wordpress-mobile/WordPress-FluxC-Android
# Woo/fetch product tags

Fixes #1611 by adding support to fetch product tags for a given site.

#### Changes
- Adds a new endpoint `wc/v3/products/tags/` to fetch product tags for a site.
- Added new `WCProductTagModel` model class and a migration script to create table.
- Added new action and action classes to `WCProductStore` and `ProductRestClient` to fetch product tags for a site.
- Added unit tests.
- Added a new button to the example app to fetch product tags.

#### Screenshots
<img src="https://user-images.githubusercontent.com/22608780/85400733-85931a00-b576-11ea-8ae2-76c2ec72dc63.png" width="300"/>

#### To test
- Since this PR adds a migration script, it would be nice to test updating the app.
- Run `MockedStack_WCProductsTest`, `ReleaseStack_WCProductTest` and `ProductSqlUtilsTest`.
- In the example app, click on `Woo` -> `Products` - > `Select Site` - > `Fetch product tags`.
  - Verify that the categories are fetched correctly.
  - Click on `Load more Tags` and verify that the tags are fetched correctly.
  - If there is no more categories data found in the API response, the `Load more tags` button should NOT be displayed.
